src/python/ensembl/io/genomio/fasta/compare.py

```diff
@@ -0,0 +1,246 @@
+
+        Returns:
+            dict: A dictionary where keys are sequence IDs and values are sequences
+              with all non-CGTA characters replaced by 'N'.
+        """
+        print(f"Read file {fasta_path}")
+        sequences = {}
+        with open_gz_file(fasta_path) as fasta_fh:
+            for rec in SeqIO.parse(fasta_fh, "fasta"):
+                name = rec.id
+                sequences[name] = re.sub(r"[^CGTA]", "N", str(rec.seq.upper()))
+        return sequences
+
+    def build_seq_dict(self, seqs: dict) -> dict:
+        """
+        Builds a dictionary of unique sequences and their associated IDs, accounting for duplicates.
+
+        Args:
+            seqs (dict): A dictionary where keys are sequence IDs and values are sequences.
+
+        Returns:
+            dict: A dictionary where keys are unique sequences and values are `SeqGroup` objects
+                that group sequence IDs sharing the same sequence.
+        """
+        seqs_dict: Dict[str, SeqGroup] = dict()
+        for name, seq in seqs.items():
+            if seq in seqs_dict:
+                seqs_dict[seq].add_id(name)
+            else:
+                seqs_dict[seq] = SeqGroup(name)
+
+        return seqs_dict
+
+    def find_common_groups(self, seq1_dict: dict, seq2_dict: dict) -> Tuple[dict, List[Any]]:
+        """
+        Find common sequences between two dictionaries and group them.
+
+        Args:
+            seq1_dict (dict): Dictionary of sequences from the first dataset.
+            seq2_dict (dict): Dictionary of sequences from the second dataset.
+
+        Returns:
+            Tuple[dict, List[str]]: A dictionary of common sequence mappings and
+                                a list of comparison results.
+        """
+        common = {}
+
+        for seq1, group1 in seq1_dict.items():
+            if seq1 in seq2_dict:
+                group2 = seq2_dict[seq1]
+                # Check that the 2 groups have the same number of sequences
+                if group1.count == group2.count:
+                    if group1.count == 1:
+                        common[group1.ids[0]] = group2.ids[0]
+                    else:
+                        self.comp.append(f"Matched 2 identical groups of sequences: {group1} and {group2}")
+                        # Map each ID in group1 to a possible group2 ID
+                        possible_id2 = " OR ".join(group2.ids)
+                        common.update({id1: possible_id2 for id1 in group1.ids})
+                else:
+                    self.comp.append(
+                        f"Matched 2 different groups of sequences ({group1.count} vs {group2.count}): {group1} and {group2}"
+                    )
+                    
+        return common, self.comp
+
+    def write_results(self) -> None:
+        """
+        Write the comparison results to a file in the output directory.
+
+        """
+        output_file = Path.joinpath(self.output_dir, "compare.log")
```

Comment:
```suggestion
        output_file = Path.joinpath(self.output_dir, "compare.log")
        observed_compare = set()
```

Reply:
Added an observed set variable to remove the repeating log outputs occurring

Reply:
sorry didnt realise, there were repeats. I will check this, thank you!